PxWeb/Code/Api2/DataSource/Cnmm/SqlDbConfigExtensions.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using PCAxis.Sql;
+using PCAxis.Sql.DbClient;
+using PCAxis.Sql.DbConfig;
+
+namespace PxWeb.Code.Api2.DataSource.Cnmm
+{
+    public static class SqlDbConfigExtensions
+    {
+        public static Dictionary<string, string> GetMenuLookup(this SqlDbConfig DB, string language)
+        {
+            //TODO: Handle language
+
+
+
+            var menuLookup = new Dictionary<string, string>();
+
+            string sql;
+            if (DB is SqlDbConfig_21)
+            {
+                sql = GetMenuLookupQuery2_1(DB as SqlDbConfig_21);
+            }
+            else if (DB is SqlDbConfig_22)
+            {
+                sql = (DB as SqlDbConfig_22).GetMenuLookupQuery2_2();
+            }
+            else if (DB is SqlDbConfig_23)
+            {
+                sql = (DB as SqlDbConfig_23).GetMenuLookupQuery2_3();
+            }
+            else if (DB is SqlDbConfig_24)
+            {
+                sql = (DB as SqlDbConfig_24).GetMenuLookupQuery2_4();
+            }
+            else
+            {
+                return null;
+            }
+
+            var cmd = GetPxSqlCommand(DB);
+
+            var dataSet = cmd.ExecuteSelect(sql);
+
+            foreach (DataRow row in dataSet.Tables[0].Rows)
+            {
+                string key = row[1].ToString().ToUpper();
+
+                // TODO: Is this ok? What happens if the table/node are present on many nodes?
```

Comment:
It is ok since the chaild nodes are always equal. TODO might be removed

Reply:
Fixed